### PR TITLE
Feature/hooks

### PR DESCRIPTION
**Pull request by Aseadd**
### **Tasks are done for the requirement**

- [x] Test calculate and operate functions
- [x] Test Calculator, Quote, Navbar, App, Home components
- [x] Apply User interaction test for Calculator buttons

### CODE REVIEW

File: src/components/__test__/App.test.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+it('should render the app component', () => {
+  const wrapper = ReactDOM.createRoot(document.createElement('div'));
+  wrapper.render(<App />);
+});
+
+it('should match the snapshot', () => {
+  const tree = renderer.create(<App />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (tresorsawasawa) [inline src/components/__test__/App.test.js:14]: - [x] Kindly make sure all the functionalities & scenarios of the `operate.js` and `calculate.js` modules should be covered as required for the project.

File: src/components/__test__/quotes.test.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+it('should render the quotes component', () => {
+  const wrapper = ReactDOM.createRoot(document.createElement('div'));
+  wrapper.render(<Quotes />);
+});
+
+it('should match the snapshot', () => {
+  const tree = renderer.create(<Quotes />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (tresorsawasawa) [inline src/components/__test__/quotes.test.js:14]: - [x] Also make sure there is the usage of **React Testing Library** to simulate user interaction for as required for the project.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Aseadd 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/__test__/App.test.js (hunk 1 above)
Comment (ErikStoupignan) [reply src/components/__test__/App.test.js:14]: Ok, we already add more testing & scenarios. Thank you for the advice.

File: src/components/__test__/quotes.test.js (hunk 2 above)
Comment (ErikStoupignan) [reply src/components/__test__/quotes.test.js:14]: Hello code reviewer! 
In this file "Quote page", we can't apply any user interaction. Quote page data come from an API, it doesn't have user interaction.

We use the user Interaction and the React Testing Library on the "Calculator page" simulating the clicks on the buttons and looking at the answers on the screen.

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi Team, 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**